h_malloc.c

```diff
@@ -528,7 +528,7 @@ static void set_canary(UNUSED const struct slab_metadata *metadata, UNUSED void
     }
 #endif
 
-    memcpy((char *)p + size - canary_size, &metadata->canary_value, canary_size);
+    h_memcpy_real((char *)p + size - canary_size, &metadata->canary_value, canary_size);
```

Comment:
maybe introduce `internal_memcpy()` as a macro and use it here and then define it depending on `CONFIG_BLOCK_OPS_CHECK_SIZE` to either `h_memcpy_real()` or `memset()`